src/components/filter/EventsFilterPanel.tsx

```diff
@@ -45,60 +53,79 @@ function EventsFilterPanel() {
 	const eventsStore = useWorkspaceEventStore();
 	const eventDataStore = useEventsDataStore();
 	const filterStore = useEventsFilterStore();
+	const { autocompletes, saveAutocompletes } = useFilterAutocompletesStore();
+	const { addHistoryItem } = useFiltersHistoryStore();
 
 	const [showFilter, setShowFilter] = React.useState(false);
 
-	const [filter, setFilter] = React.useState<EventFilterState | null>(filterStore.filter);
 	const [currentFilterValues, setCurrentFilterValues] = React.useState<CurrentFilterValues | null>(
 		getDefaultCurrentFilterValues(filterStore.filter),
 	);
 
 	React.useEffect(() => {
-		setFilter(filterStore.filter);
 		setCurrentFilterValues(getDefaultCurrentFilterValues(filterStore.filter));
 	}, [filterStore.filter]);
 
 	const onSubmit = React.useCallback(() => {
-		if (filter) {
-			eventsStore.applyFilter(filter);
+		if (filterStore.filter) {
+			// eslint-disable-next-line @typescript-eslint/no-unused-vars
+			const { status, ...restFilters } = filterStore.filter;
+			const filtersToSave: FiltersToSave = Object.fromEntries(
+				Object.entries(restFilters)
+					// eslint-disable-next-line @typescript-eslint/no-unused-vars
+					.filter(([_, value]) => value.values.length > 0)
+					.map(([key, value]) => [
+						key,
+						toJS(typeof value.values === 'string' ? value.values : value.values.sort()),
```

Comment:
this throwns warning in console as you trying to sort observable array, first use convert to js